core/ir/decode_shared.c

```diff
@@ -253,8 +253,11 @@ dr_set_isa_mode(void *drcontext, dr_isa_mode_t new_mode,
     dcontext_t *dcontext = (dcontext_t *)drcontext;
     dr_isa_mode_t old_mode;
     /* We would disallow but some early init routines need to use global heap */
-    if (dcontext == GLOBAL_DCONTEXT)
+    if (dcontext == GLOBAL_DCONTEXT) {
         dcontext = get_thread_private_dcontext();
+        if (dcontext == NULL)
```

Comment:
Maybe add a comment like: if thread private dcontext is not ready (pre-thread init), fall back on GLOBAL_DCONTEXT.
(If I understand this correctly.)

Reply:
Done.